Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -1442,7 +1467,9 @@ void TextEditor::insert_at_cursor_or_replace_selection(StringView text)
         && clear_length > 0
         && current_line().leading_spaces() == clear_length;
 
-    execute<InsertTextCommand>(text, m_cursor);
+    auto const should_insert_text = (has_selection() && text != "\t") || !has_selection();
```

Comment:
Not sure I understand this actually. What are we checking text against being a single tab character, and why is it relevant to inserting text?

Reply:
In case this is to represent 'single tab press': that seems like a really awkward way of doing it, and definitely deserves a comment.

Reply:
Ya, this code was a mess. What was happening was that when we indent a selection, it would indent the selection but also insert a tab wherever the selection started, which we don't want. So I tried to write in some conditionals here to fix that, but it didn't really work and as you saw was really unclear. I've now added a function to check whether we are indenting the selection which can be used here and is more clear. Let me know if it makes sense now, thanks :^)